woocommerce/woocommerce
# dev: bump pnpm version to 9.15.0

- **bumped pnpm version to 9.15.0**
- **pnpm lock update**

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/50403

With the work done by Chi in https://github.com/woocommerce/woocommerce/pull/52815 it becomes possible to update the pnpm version without causing build breakage. 

This PR bumps the pnpm version to the latest 9.15.0. This supersedes a previous PR that bumped it to 9.3.0, as that PR caused a workflow breakage due to it checking out earlier versions of the codebase while setting up the CI environment with pnpm 9.3.0 - causing a pnpm version mismatch.

Note that as of pnpm 9.7.0, pnpm will automatically down-version to match the `packageManager` field, so the breakage above should not happen anymore.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Observe that build and CI passes.



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Bump pnpm version across the whole repo to 9.15.0

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
